src/plugins/explore/public/application/utils/state_management/utils/redux_persistence.ts

```diff
@@ -246,8 +247,13 @@ const getPreloadedQueryEditorState = async (
   queryState?: QueryState
 ): Promise<QueryEditorSliceState> => {
   let promptModeIsAvailable = false;
+  let summaryAgentIsAvailable = false;
   if (queryState?.dataset) {
     promptModeIsAvailable = await getPromptModeIsAvailable(services);
+    summaryAgentIsAvailable = await getSummaryAgentIsAvailable(
+      services,
+      queryState.dataset.dataSource?.id!
```

Comment:
1. lets use promise.allSetttled for above two
2. what hapepns if dataSource? or id?